compiler/src/main/java/org/qbicc/type/definition/classfile/ClassFileImpl.java

```diff
@@ -345,6 +352,14 @@ public Literal getConstantValue(int idx, TypeParameterContext paramCtxt) {
         }
     }
 
+    private ValueType getReferencedObjectType(ReferenceType refType) {
+        if (refType.getInterfaceBounds().size() > 0) {
```

Comment:
Just a note that this only works because in this particular case you'll never have a type with both a class bound and an interface bound, and you'll never have a type with more than one interface bound.  At least, I think that is the case. In any event it's better not to have this logic at all.